src/ndarray/ndarray_function-inl.h

```diff
@@ -47,6 +47,15 @@
   }
 #endif
 
+#ifndef DECL_BINARY_LAUNCH
+#define DECL_BINARY_LAUNCH(XPU, OP)                                               \
```

Comment:
why not just put BinaryOpKernelLaunch in header?

Reply:
This follows same style as existing code like Eval.

Are you referring to putting BinaryOpKernelLaunch in header or putting the definition of BinaryOpKernelImpl in ndarray_function.h ?